src/figuresClasses.ts

```diff
@@ -1,19 +1,87 @@
-export interface Figure {
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
 
+enum Color {
+  Red = 'red',
+  Green = 'green',
+  Blue = 'blue',
 }
 
-export class Triangle {
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
 
+function checkSizes(...sizes: number[]): void {
+  if (!sizes.every((size) => size > 0)) {
+    throw new Error('Initial size should be more than 0.');
+  }
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  shape = Shape.Triangle;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    checkSizes(a, b, c);
+
+    if (
+      a + b <= c
+      || a + c <= b
+      || b + c <= a
+    ) {
+      throw new Error('Wrong sizes for triangle sides.');
+    }
+  }
 
+  getArea(): number {
+    const semiPerimeter = (this.a + this.b + this.c) / 2;
+
+    return Math.sqrt(semiPerimeter * (semiPerimeter - this.a)
+      * (semiPerimeter - this.b) * (semiPerimeter - this.c));
+  }
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  shape = Shape.Circle;
+
+  constructor(
+    public color: Color,
+    public radius: number,
+  ) {
+    checkSizes(radius);
+  }
 
+  getArea(): number {
+    return 3.14 * (this.radius ** 2);
+  }
 }
 
-export function getInfo(figure) {
+export class Rectangle implements Figure {
+  shape = Shape.Rectangle;
+
+  constructor(
+    public color: Color,
+    public width: number,
+    public height: number,
+  ) {
+    checkSizes(width, height);
+  }
+
+  getArea(): number {
+    return this.width * this.height;
```

Comment:
I guess you should round down to hundreds all areas of the figures.
If we change input values we can get results like 45,3333333

Reply:
I did it in getInfo function